plugins/modules/python_requirements_info.py

```diff
@@ -165,7 +166,7 @@ def main():
             python_version_info=python_version_info,
             python_system_path=sys.path,
         )
-    pkg_dep_re = re.compile(r'(^[a-zA-Z][a-zA-Z0-9_-]+)(?:(==|[><]=?)([0-9.]+))?$')
+    pkg_dep_re = re.compile(r'(^[a-zA-Z][a-zA-Z0-9_-]+)(?:\[([a-z][a-zA-Z0-9_ ]+)\])?(?:(==|[><]=?)([0-9.]+))?$', flags=re.IGNORECASE)
```

Comment:
A couple of issues here:
* Since using `re.IGNORECASE` we can reduce `[a-zA-Z]` to `[a-z]`
* In the extras part is missing:
  * If I got it right in https://peps.python.org/pep-0508/ it says that extras can begin with digits as well (very very weird to me, but that seems to be rule)
  * Optional whitespaces in the beginning and the end
  * Multiple extras separated by commas
* Optional whitespaces around version numbers
* Multiple version operators+operands (as in the example in the next line)
* Missing the extra markers (to be fair that was not in the issue ticket), as in `requests [security,tests] >= 2.8.1, == 2.8.* ; python_version < "2.7"` the part including and after the semi-colon 

Trying to solve part of these issues:
```suggestion
    pkg_dep_re = re.compile(r'(^[a-z][a-z0-9_-]+) *(?:\[ *([a-z0-9][a-z0-9_ ]+) *\])? *(?:(==|[><]=?) *([0-9.]+))?$', flags=re.IGNORECASE)
```
We might want to raise another PR in the future to complete the specification.

Reply:
Hello Russoz,

I've seen and read the PEP-508 with the hope to fine detailed implementation but their parsing does not seem to be regex based (and I understand why when I see the complexity of the spec). Yet, the module requires [pkg_resources](https://setuptools.pypa.io/en/latest/pkg_resources.html) to work and I am wondering if it does not support the PEP-508 already in which case the actual code should probably be only updated to not filter the input but catch possible **RequirementParseError** on top of the existing DistributionNotFound exception that already exist.